spec.html

```diff
@@ -38,6 +38,216 @@ <h1>Well-known intrinsic objects</h1>
+			WrappedModuleNamespaceCreate (
+				_callerRealm_: a Realm Record,
+				_module_: a Module Record,
+				_exports_: a List of Strings,
+			): a wrapped module namespace exotic object
+		</h1>
+		<dl class="header">
+			<dt>description</dt>
+			<dd>It is used to specify the creation of new wrapped module namespace exotic objects.</dd>
+		</dl>
+		<emu-alg>
+			1. Let _internalSlotsList_ be the internal slots listed in <emu-xref href="#table-internal-slots-of-wrapped-module-namespace-exotic-objects"></emu-xref>.
+			1. Let _M_ be MakeBasicObject(_internalSlotsList_).
+			1. Set _M_'s essential internal methods to the definitions specified in <emu-xref href="#sec-wrapped-module-namespace-exotic-objects"></emu-xref>.
+			1. Set _M_.[[Module]] to _module_.
+			1. Let _sortedExports_ be a List whose elements are the elements of _exports_ ordered as if an Array of the same values had been sorted using %Array.prototype.sort% using *undefined* as _comparefn_.
+			1. Set _M_.[[Exports]] to _sortedExports_.
+			1. Set _M_.[[Realm]] to _callerRealm_.
+			1. Create own properties of _M_ corresponding to the definitions in <emu-xref href="#sec-wrapped-module-namespace-exotic-objects"></emu-xref>.
+			1. Return _M_.
+		</emu-alg>
+	</emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-finishshadowrealmimport" type="abstract operation">
+	<h1>
+		FinishShadowRealmImport (
+			_callerRealm_: a Realm Record,
+			_specifier_: unknown,
+			_promiseCapability_: a PromiseCapability Record,
+			_innerPromise_: unknown,
+		): ~unused~
+	</h1>
+	<dl class="header">
+		<dt>description</dt>
+		<dd>FinishShadowRealmImport completes the process of importing a module into a ShadowRealm, resolving or rejecting the promise returned by that call as appropriate according to _innerPromise_'s resolution. It is performed by host environments as part of HostImportValueModuleDynamically.</dd>
+	</dl>
+	<emu-alg>
+		1. Let _fulfilledClosure_ be a new Abstract Closure in the context of _callerRealm_ with parameters (_result_) that captures _specifier_, and _promiseCapability_ and performs the following steps when called:
+			1. Assert: _result_ is *undefined*.
+			1. Let _runningContext_ be the running execution context.
+			1. If _runningContext_ is not already suspended, suspend _runningContext_.
+			1. Push _evalContext_ onto the execution context stack; _evalContext_ is now the running execution context.
+			1. Let _moduleRecord_ be ! HostResolveImportedModule(*null*, _specifier_).
+			1. Assert: Evaluate has already been invoked on _moduleRecord_ and successfully completed.
+			1. Let _namespace_ be Completion(GetModuleNamespace(_moduleRecord_)).
+			1. Suspend _evalContext_ and remove it from the execution context stack.
+			1. Resume the context that is now on the top of the execution context stack as the running execution context.
+			1. If _namespace_ is an abrupt completion, then
+				1. Let _error_ be an Error object from _callerRealm_ equivalent to _namespace_.[[Value]].
```

Comment:
IMPORTANT: I could not figure (from the latest 262 specs) when is this going to occur and what the value of `_namespace_.[[Value]]` will be under those circumstances. In the old days, if there was an abrupt completion when getting the namespace, it's `[[Value]]` was never set IIRC. I need some help here.

Ultimate, the goal here is to produce an error object in the `_callerRealm_` that can have as much information as possible about the actual error if the error is related to the module graph resolution mechanisms.

Reply:
It might very well be, but I'm not sure @guybedford

Reply:
Would be worth verifying again for certain, but I'm pretty much 90% sure that is the case.